### PR TITLE
Feature - Edit Profile Form

### PR DESCRIPTION
**Pull request by jaypeasee**
# Pull Request Template

#### What’s this PR do? 
1. Builds a new component called `EditProfile` that is linked from `Profile`
2. Creates local state in `EditProfile` for `image`, `name`, `email`, `bio`, `street1`, `street2`, `city`, `state` and `zip`.
3. Routes `EditProfile` back to `Profile` on form submit
4. Adds/updates `user` `details` in Redux store with `image`, `name`, `email`, `bio`,  `role`, and `streetAddress`, using the `editProfile()` method and `EDIT_PROFILE` action on form submit.

#### Where should the reviewer start?
Review `Profile.js`, `EditProfile.js`, `actions.js`, and `state.js` in `web/src/modules/user`

#### How should this be manually tested?
1. Login into a user profile
2. Click `[EDIT PROFILE]` to redirect to the edit profile form.
3. Fill out the form. Check local state in React Dev Tools
4. Submit the form. Check that you redirect to the Profile, and the Redux dev tools updates the store of user details.

#### Next Steps
* Confirm required fields in the form
* Make either a PATCH or PUT request to the api the backend sends
* Create conditions to populate the field if state already exists for it.

#### What are the relevant tickets? 
#31 
#33 

#### Screenshots (if appropriate): 
Screen Shot 2021-02-06 at 8 27 32 PM

### CODE REVIEW

File: code/web/src/modules/user/EditProfile.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    }
+};
+
+function editProfileState(state) {
+  return {
+    user: state.user,
+  };
+}
+
+// export default EditProfile
```
Comment (ADavidson02) [inline code/web/src/modules/user/EditProfile.js:258]: We need to delete this unneeded code.

**Review by ADavidson02 — APPROVED** [review]
Looks good to me.

**Review by kevxo — COMMENTED** [review]
Looks good to me

**Comment by carson-jardine** [comment]
This looks good to me! I'd love to walk through it to get a better sense of how the React Dev Tools works! 
Also, I like the PR Template you used, should we update our current PR Template to this one?

**Review by carson-jardine — APPROVED** [review]

**Pull request merged**